src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Infer.java

```diff
@@ -213,6 +213,17 @@ Type instantiateMethod( Env<AttrContext> env,
 
             deferredAttrContext.complete();
 
+            /* if boxing is not allowed then, no undetVar should have a bound that is a primitive class, unless the bound
```

Comment:
Looks good and very minimal - probably a good starting option for a prototype. Other approaches I thought of include setting "poison" bounds on the inference variable, so as to make subtyping/equality/containment fail.  But it's more complex and less localized.

I believe this might just work. Of course some tests are needed :-)

Reply:
yep, I will add more tests, thanks for the review